pkg/occlient/occlient_test.go

```diff
@@ -406,3 +408,883 @@ func Test_parseImageName(t *testing.T) {
+			dcName:      "wildfly",
+			projectName: "project-testing",
+			annotations: map[string]string{
+				"app.kubernetes.io/url":                   "file:///temp/nodejs-ex",
+				"app.kubernetes.io/component-source-type": "local",
+			},
+			labels: map[string]string{
+				"app": "apptmp",
+				"app.kubernetes.io/component-name": "ruby",
+				"app.kubernetes.io/component-type": "ruby",
+				"app.kubernetes.io/name":           "apptmp",
+			},
+			existingDc: appsv1.DeploymentConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "wildfly",
+					Annotations: map[string]string{
+						"app.kubernetes.io/url":                   "https://github.com/sclorg/nodejs-ex",
+						"app.kubernetes.io/component-source-type": "git",
+					},
+				},
+				Spec: appsv1.DeploymentConfigSpec{
+					Template: &corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"deploymentconfig": "wildfly",
+							},
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Image: "wildfly:latest",
+									Name:  "wildfly",
+								},
+							},
+						},
+					},
+				},
+			},
+			createdPVC: corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: fmt.Sprintf("%s-s2idata", "wildfly"),
+					Labels: map[string]string{
+						"app": "apptmp",
+						"app.kubernetes.io/component-name": "wildfly",
+						"app.kubernetes.io/component-type": "wildfly",
+						"app.kubernetes.io/name":           "apptmp",
+					},
+				},
+				Spec: corev1.PersistentVolumeClaimSpec{
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: errQuantity,
+						},
+					},
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+				},
+			},
+
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fkclient, fkclientset := FakeNew()
+			fkclientset.AppsClientset.PrependReactor("get", "deploymentconfigs", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				if tt.dcName != tt.existingDc.Name {
+					return true, nil, fmt.Errorf("got different dc")
+				}
+				return true, &tt.existingDc, nil
+			})
+
+			fkclientset.AppsClientset.PrependReactor("update", "deploymentconfigs", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				dc := action.(ktesting.UpdateAction).GetObject().(*appsv1.DeploymentConfig)
+				if dc.Name != tt.existingDc.Name {
+					return true, nil, fmt.Errorf("got different dc")
+				}
+				return true, dc, nil
+			})
+
+			fkclientset.Kubernetes.PrependReactor("create", "persistentvolumeclaims", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				createdPvc := action.(ktesting.CreateAction).GetObject().(*corev1.PersistentVolumeClaim)
+				if createdPvc.ObjectMeta.Name != tt.createdPVC.ObjectMeta.Name {
```

Comment:
hmm, I am not sure if we need these checks. If you are just checking the values that you have provided to the reactor and its output. Not really checking our function. @kadel  WDYT?

Reply:
@ashetty1 Checking if the created PVC is created with the required name or not